From 122a660be74104202f815c8fe73a88aba2560c3c Mon Sep 17 00:00:00 2001
From: Peter Meerwald <p.meerwald@bct-electronic.com>
Date: Wed, 25 Apr 2012 19:36:36 +0200
Subject: [PATCH] beagle: fix 'timed out in wait_for_bb' message in SPL

SPL boot outputs a 'timed out in wait_for_bb: IC2_STAT' message on beagle

U-Boot SPL 2012.04-00020-gb8310b9-dirty (Apr 25 2012 - 18:49:57)
Texas Instruments Revision detection unimplemented
OMAP SD/MMC: 0
timed out in wait_for_bb: I2C_STAT=1000
reading u-boot.img

the reason for above message is that when booting from MMC, I2C needs (?) to be
initialized as well

when SPL initializes MMC (which is done in omap_hsmmc.c, mmc_board_init()) the following
is called:

        twl4030_power_mmc_init();

in order to communicate with the twl4030, I2C is necessary, but I2C has not been initialized yet in SPL

the problem can be easily fixed by #defining CONFIG_SPL_BOARD_INIT in include/configs/omap3_beagle.h

tested on beagle-xm (rev. C)

Signed-off-by: Peter Meerwald <p.meerwald@bct-electronic.com>
---
 include/configs/omap3_beagle.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/omap3_beagle.h b/include/configs/omap3_beagle.h
index ddeb414..27e51cd 100644
--- a/include/configs/omap3_beagle.h
+++ b/include/configs/omap3_beagle.h
@@ -410,6 +410,7 @@
 #define CONFIG_SYS_MMC_SD_FAT_BOOT_PARTITION	1
 #define CONFIG_SPL_FAT_LOAD_PAYLOAD_NAME	"u-boot.img"
 
+#define CONFIG_SPL_BOARD_INIT
 #define CONFIG_SPL_LIBCOMMON_SUPPORT
 #define CONFIG_SPL_LIBDISK_SUPPORT
 #define CONFIG_SPL_I2C_SUPPORT
-- 
1.7.10

